hal/halx86/smp/smp.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Core source file for SMP management
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <hal.h>
+#include <smp.h>
+#include "smpp.h"
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+extern PHYSICAL_ADDRESS HalpLowStubPhysicalAddress;
+extern PVOID HalpLowStub;
+
+/* TODO: MaxAPCount should be assigned by a Multi APIC table */
+ULONG MaxAPCount = 2;
+ULONG StartedProcessorCount = 1;
+
+/* FUNCTIONS *****************************************************************/
+
+BOOLEAN
+NTAPI
+HalStartNextProcessor(
+    IN PLOADER_PARAMETER_BLOCK LoaderBlock,
+    IN PKPROCESSOR_STATE ProcessorState)
+{
+    if (MaxAPCount > StartedProcessorCount)
+    {
+        /* We only a create a new pagetable once */
+        if (StartedProcessorCount == 1)
+            HalpInitializeAPPageTables(HalpLowStub);
```

Comment:
This shouldn't be done here, but in the achitecture dependent code in ntoskrnl: You allocate a page, map it, copy the lower half of the current page table root, unmap it, store the PFN in ProcessorState.SpecialRegister.Cr3.

Reply:
Interesting... Okay ill remove this stub in this PR, and we can touch on this subject again over PM? before i submit the next one. Id really like to try this idea